### PR TITLE
Starterkit and Approvals Grid

### PR DESCRIPTION
**Pull request by skilledDev96**
Fixed Viewing of Starterkit Docs
Fixed Starterkit Docs Dissapearing and Showing N/A Fixed Filtering on Approvals Grid

### CODE REVIEW

File: src/app/components/hris/employees/employee-approvals/employee-approvals.component.ts [hunk 1]
```diff
@@ -111,15 +119,16 @@ export class EmployeeApprovalsComponent {
...
-
-          if (documentsForEmployee.length < 4) {
-            if (selectedStatus === 1 || selectedStatus === 2) {
+          //If there are less than 3 documents uploaded for an employee, Status is pending.
+          if (documentsForEmployee.length < this.MIN_STARTERKIT_DOCUMENTS_UPLOADED) {
+            if (selectedStatus === this.PENDING ) {
               this.filterDocumentTypeAndStatus(currentDto, false, selectedStatus);
             }
           } else {
+            //Statusses for Approved and Declined will only be applicable if all Documents Uploaded match the same status.
```
Comment (Juanro-Minne) [inline src/app/components/hris/employees/employee-approvals/employee-approvals.component.ts:128]: You might have to remove comments

File: src/app/components/hris/employees/employee-approvals/employee-approvals.component.ts [hunk 2]
```diff
@@ -111,15 +119,16 @@ export class EmployeeApprovalsComponent {
             indexVisitedArray.push(id);
             documentsForEmployee.push(this.bankingAndStarterKitData[id].employeeDocumentDto);
           });
-
-          if (documentsForEmployee.length < 4) {
-            if (selectedStatus === 1 || selectedStatus === 2) {
+          //If there are less than 3 documents uploaded for an employee, Status is pending.
```
Comment (Juanro-Minne) [inline src/app/components/hris/employees/employee-approvals/employee-approvals.component.ts:122]:
```suggestion

```

File: src/app/components/hris/employees/employee-approvals/employee-approvals.component.ts [hunk 3]
```diff
@@ -93,16 +97,20 @@ export class EmployeeApprovalsComponent {
...
     for (let i = 0; i < this.bankingAndStarterKitData.length; i++) {
       if (!indexVisitedArray.includes(i)) {
         const currentDto = this.bankingAndStarterKitData[i];
 
+        //BANKING
         if (currentDto.employeeBankingDto && currentDto.employeeBankingDto.status === selectedStatus) {
           this.filterDocumentTypeAndStatus(currentDto, true, selectedStatus);
         }
 
+        //STARTERKITS
```
Comment (Juanro-Minne) [inline src/app/components/hris/employees/employee-approvals/employee-approvals.component.ts:113]:
```suggestion
     
```

File: src/app/components/hris/employees/employee-approvals/employee-approvals.component.ts [hunk 4]
```diff
@@ -93,16 +97,20 @@ export class EmployeeApprovalsComponent {
...
-    const selectedStatus = selectedTabIndex === 2 ? 2 : selectedTabIndex === 1 ? 0 : 1;
+    const selectedStatus = selectedTabIndex === 0 ? this.PENDING : 
+                           selectedTabIndex === 1 ? this.APPROVED : 
+                                                    this.DECLINED ;
 
     for (let i = 0; i < this.bankingAndStarterKitData.length; i++) {
       if (!indexVisitedArray.includes(i)) {
         const currentDto = this.bankingAndStarterKitData[i];
 
+        //BANKING
```
Comment (Juanro-Minne) [inline src/app/components/hris/employees/employee-approvals/employee-approvals.component.ts:108]:
```suggestion
   
```

File: src/app/components/hris/employees/employee-approvals/employee-approvals.component.ts (hunk 1 above)
Comment (Juanro-Minne) [inline src/app/components/hris/employees/employee-approvals/employee-approvals.component.ts:128]:
```suggestion
       
```

File: src/app/components/hris/employees/employee-approvals/employee-approvals.component.ts [hunk 5]
```diff
@@ -43,6 +43,10 @@ export class EmployeeApprovalsComponent {
   screenWidth: number = 992;
   dataSource: MatTableDataSource<any> = new MatTableDataSource();
   dialogTypeData!: Dialog;
+  APPROVED = 0;
```
Comment (tbddorothy) [inline src/app/components/hris/employees/employee-approvals/employee-approvals.component.ts:46]: Rather make these into enums in a separate file

File: src/app/services/hris/employee/employee-bankingandstarterkit.service.ts [hunk 6]
```diff
@@ -14,7 +14,7 @@ export class EmployeeBankingandstarterkitService {
   readonly APPROVED_STATUS = 0;
   readonly PENDING_STATUS = 1;
```
Comment (tbddorothy) [inline src/app/services/hris/employee/employee-bankingandstarterkit.service.ts:15]: Repurpose the enum here

File: src/app/components/hris/employees/employee-approvals/employee-approvals.component.ts (hunk 5 above)
Comment (skilledDev96) [reply src/app/components/hris/employees/employee-approvals/employee-approvals.component.ts]: Created Enum file

File: src/app/services/hris/employee/employee-bankingandstarterkit.service.ts (hunk 6 above)
Comment (skilledDev96) [reply src/app/services/hris/employee/employee-bankingandstarterkit.service.ts]: Used enums created.

**Review by tbddorothy — APPROVED** [review]

**Pull request merged**